ai_feedback/__main__.py

```diff
@@ -197,32 +208,25 @@ def main() -> int:
     else:
         request, response = code_processing.process_code(args, prompt_content)
 
-    if args.output == "markdown":
-        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
-        assignment_directory = f"{TEST_OUTPUTS_DIRECTORY}/{args.model}"
-        os.makedirs(assignment_directory, exist_ok=True)
-
-        markdown_filename = f"{assignment_directory}/{args.prompt}_{timestamp}.md"
-
-        markdown_template = load_markdown_template()
-        markdown_output = markdown_template.format(
-            question=args.question if args.question else "N/A",
-            model=args.model,
-            request=request,
-            response=response,
-            timestamp=timestamp,
-            submission=args.submission,
-        )
-        with open(markdown_filename, "w") as md_file:
-            md_file.write(markdown_output)
-        print(f"Markdown report saved to `{markdown_filename}`")
-
-    elif args.output == "stdout":
-        print(response)
-
-    elif args.output == "direct":
-        print(response)
+    markdown_template = load_markdown_template(args.output_template)
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+    output_text = markdown_template.format(
+        question=args.question or "N/A",
+        model=args.model,
+        request=request,
+        response=response,
+        timestamp=timestamp,
+        submission=args.submission,
+    )
 
+    if args.output:
+        output_path = Path(args.output)
+        output_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(output_path, "w") as f:
+            f.write(output_text)
+        print(f"Markdown report saved to `{output_path}`")
```

Comment:
This isn't a bad idea, but is unconventional; it's more conventional to print nothing, as if this program was redirecting the output (`ai_feedback > my_output_file.md`)